asreview/webapp/api/auth.py

```diff
@@ -0,0 +1,512 @@
+            result = (500, f"Unable to reset your password! Reason: {str(e)}")
+        except SQLAlchemyError as e:
+            DB.session.rollback()
+            result = (500, f"Unable to reset your password! Reason: {str(e)}")
+
+    status, message = result
+    response = jsonify({"message": message})
+    return response, status
+
+
+@bp.route("/refresh", methods=["GET"])
+def refresh():
+    if current_user and isinstance(current_user, User):
+        logged_in = current_user.is_authenticated
+        name = current_user.get_name()
+        id = current_user.id
+    else:
+        logged_in = False
+        name = ""
+        id = None
+
+    result = {"logged_in": logged_in, "name": name, "id": id}
+
+    response = jsonify(result)
+    return response, 200
+
+
+@bp.route("/signout", methods=["DELETE"])
+@asreview_login_required
+def signout():
+    if current_user:
+        identifier = current_user.identifier
+        logout_user()
+        result = (200, f"User with identifier {identifier} has been signed out")
+    else:
+        result = (404, "No user found, no one can be signed out")
+
+    status, message = result
+    response = jsonify({"message": message})
+    return response, status
+
+
+@bp.route("/oauth_callback", methods=["POST"])
+def oauth_callback():
+    # get parameters
+    code = request.form.get("code", "").strip()
+    provider = request.form.get("provider", "").strip()
+    redirect_uri = request.form.get("redirect_uri", "").strip()
+
+    # assuming we have this provider
+    oauth_handler = current_app.config.get("OAUTH", False)
+    if (
+        isinstance(oauth_handler, OAuthHandler)
+        and provider in oauth_handler.providers()
+    ):
+        # get user credentials for this user
+        response = oauth_handler.get_user_credentials(provider, code, redirect_uri)
+        (identifier, email, name) = response
+        # try to find this user
+        user = User.query.filter(User.identifier == identifier).one_or_none()
+        # flag for response (I'd like to communicate if this user was created)
+        created_account = False
+        # if not create user
+        if user is None:
+            try:
+                origin = provider
+                confirmed = True
+                public = True
+                user = User(
+                    identifier=identifier,
+                    origin=origin,
+                    email=email,
+                    name=name,
+                    confirmed=confirmed,
+                    public=public,
+                )
+                DB.session.add(user)
+                DB.session.commit()
+                created_account = True
+            except SQLAlchemyError:
+                DB.session.rollback()
+                message = "OAuth: unable to create your account!"
+                # return this immediately
+                return jsonify({"data": message}), 500
+
+        # log in this user
+        if bool(user):
```

Comment:
Is there any way this can evaluate to false? We already know that user is not None.

Reply:
It's just me being paranoid, I have removed the if statement.